manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml

```diff
@@ -41,29 +43,57 @@ webhooks:
         operator: DoesNotExist
       - key: istio.io/rev
         operator: DoesNotExist
-{{- else if .Values.revision }}
+      {{- else }}
       matchExpressions:
       - key: istio-injection
-        operator: DoesNotExist
-      - key: istio.io/rev
         operator: In
         values:
-        - {{ .Values.revision }}
-{{- else }}
-      matchLabels:
-        istio-injection: enabled
-{{- end }}
-{{- if .Values.sidecarInjectorWebhook.objectSelector.enabled }}
+          - enabled
+      - key: istio.io/rev
+        operator: DoesNotExist
+      {{- end }}
+    {{- if .Values.sidecarInjectorWebhook.objectSelector.enabled }}
     objectSelector:
-{{- if .Values.sidecarInjectorWebhook.objectSelector.autoInject }}
+      {{- if .Values.sidecarInjectorWebhook.objectSelector.autoInject }}
       matchExpressions:
       - key: "sidecar.istio.io/inject"
         operator: NotIn
         values:
         - "false"
-{{- else }}
+      {{- else }}
       matchLabels:
         "sidecar.istio.io/inject": "true"
+    {{- end }}
+  {{- end }}
 {{- end }}
-{{- end }}
+
+{{/*
+Set up another webook that will match istio.io/rev label. If a revision is set, then we will make sure
+that we do NOT match istio-injection=enabled. This avoids injecting other revisions. If a revision is NOT
+set, we will match istio.io/rev=default. This means we also match a namespace that is (by user error)
+labeled with both istio.io/rev=default,istio-injection=enabled.
+ */}}
+  - name: sidecar-injector.istio.io
+    clientConfig:
+      service:
+        name: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
+        namespace: {{ .Release.Namespace }}
+        path: "/inject"
+      caBundle: ""
+    rules:
+      - operations: [ "CREATE" ]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+    failurePolicy: Fail
+    namespaceSelector:
+      matchExpressions:
+      {{- if ne .Values.revision ""}}
```

Comment:
It would be cleaner if this second webhook applies only to 'revision != ""', and the first webhook for '*' or old-style istio-inject=true.